src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,127 @@
+            throw new DataProcessingException("Can't create book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql);) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(setBookData(id, resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book with ID " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String sql = "SELECT * FROM books";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql);
+                ResultSet resultSet = statement.executeQuery()) {
+
+            while (resultSet.next()) {
+                Long id = resultSet.getLong("id");
+                books.add(setBookData(id, resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all books from DB", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        Long bookId = book.getId();
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setString(1, book.getTitle());
+            statement.setObject(2, book.getPrice());
+            statement.setObject(3, bookId);
+            int affectedRows = statement.executeUpdate();
+
+            if (affectedRows < 1) {
+                throw new RuntimeException(
+                        "Expected to update at least 1 row, but updated 0 row.");
+            }
+
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book with ID " + bookId, e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String sql = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+
+            statement.setLong(1, id);
+            int affectedRows = statement.executeUpdate();
+            return affectedRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete book with ID " + id, e);
+        }
+    }
+
+    private Book setBookData(Long id, ResultSet resultSet) throws SQLException {
```

Comment:
setBookData name looks like setter in model and is not obvious for person who reads the code
```suggestion
    private Book parseBookData(Long id, ResultSet resultSet) throws SQLException {
```